Assets/Plugins/iOS/Shopify/WebCheckoutSession.swift

```diff
@@ -0,0 +1,128 @@
+//
+//  WebCheckoutSession.swift
+//  UnityBuySDK
+//
+//  Created by Shopify.
+//  Copyright © 2017 Shopify Inc. All rights reserved.
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files (the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions:
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+//
+
+import Foundation
+import WebKit
+
+private var activeWebPaySession: WebCheckoutSession?
+
+protocol WebCheckoutDelegate: class {
+    func didLoadThankYouPage()
+    func willDismiss(completionHandler: UnityMessage.MessageCompletion?)
+}
+
+@objc class WebCheckoutSession: NSObject {
+    static func createSession(unityDelegateObjectName: String, url: String) -> WebCheckoutSession {
+        let session = WebCheckoutSession(unityDelegateObjectName: unityDelegateObjectName,
+                                         checkoutURL: url)
+        activeWebPaySession = session
+        return session;
+    }
+    
+    fileprivate var webViewController: Checkout.WebViewController!
+    fileprivate var overlay: UIView?
+    
+    private let checkoutURL: String
+    private let overlayAnimationDuration: TimeInterval = 0.4
+    fileprivate let unityDelegateObjectName: String
+    
+    init(unityDelegateObjectName: String, checkoutURL: String) {
+        self.unityDelegateObjectName = unityDelegateObjectName
+        self.checkoutURL = checkoutURL
+        self.overlay = {
+            let view = UIView(frame: UIScreen.main.bounds)
+            view.autoresizingMask = [.flexibleHeight, .flexibleWidth]
+            view.backgroundColor = .black
+            view.alpha = 0
+            return view
+        }()
+        
+        super.init()
+    }
+    
+    func startCheckout() -> Bool {
+        let root = UnityAppController.root
```

Comment:
Nit: This local variable seems to only be used in one place. Can just call this inline:
```swift
UnityAppController.root.present(webViewController, animated: true, completion: nil)
```